New closets

MTandi opened the pull request:
![closets_flat](https://github.com/tgstation/tgstation/assets/3625094/aa27979b-68a6-4d54-9ab5-5646281dd1fd)

![closets_flat_1](https://github.com/tgstation/tgstation/assets/3625094/04719e51-f8c8-479b-bebd-7aa8714682a8)

<details>
  <summary>In-game look</summary>

![image](https://github.com/tgstation/tgstation/assets/3625094/786f7c16-61c4-4670-bec8-3457076f58a9)

</details>

## About The Pull Request

Resprites the existing closets and adds new ones.

TODO:

- [x] Resprite cabinets
- [x] Split wardrobe cabinets into unique closets
- [x] Add new sprites for some closets that use generic style as a fallback
- [x] Update the cursed closets and fridges for the new dimensions
- [x] Bio closet for medbay
- [x] Re-map some locker icons like chemistry, anasthetic...

## Why It's Good For The Game

Making them in the same style as the crates.

## Changelog

:cl:
fix: Emag overlay on lockers fixed
image: New locker sprites
image: Added new lockers
/:cl:

MTandi committed:
initial

MTandi committed:
emag logic

MTandi committed:
icon_broken

MTandi committed:
initial version

CoiledLamb commented:
what's up with the doors not matching the lockers? it's a weird design choice once, but like, i'm seeing it 6-7 times

mc-oofert commented:
i think lockers should remain as tall as they were prior but thats your choice so

MTandi commented:
> what's up with the doors not matching the lockers? it's a weird design choice once, but like, i'm seeing it 6-7 times

Made the heads` lockers have the black base with golden hooks, and biosuit lockers to be in a similar style, while not looking like some department lockers. Please suggest how I can improve this if you think it looks weird.

MTandi commented:
> i think lockers should remain as tall as they were prior but thats your choice so

Will consider lowering them a couple pixels down.

Mothblocks commented on code/game/objects/structures/crates_lockers/closets.dm:
```diff
@@ -199,6 +200,11 @@ GLOBAL_LIST_EMPTY(roundstart_station_closets)
 		. += icon_welded
 
+	if(broken && secure)
+		. += mutable_appearance(icon, icon_broken, alpha = src.alpha)
+		. += emissive_appearance(icon, icon_broken, src, alpha = src.alpha)
```
```suggestion
		. += mutable_appearance(icon, icon_broken, alpha = alpha)
		. += emissive_appearance(icon, icon_broken, src, alpha = alpha)
```

Mothblocks reviewed (changes requested):
I really like the small height difference

MTandi committed:
Update code/game/objects/structures/crates_lockers/closets.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

Acantharctia commented:
Looks amazing, but the engineering lockers lacking a door will look the same like the example above it?

Acantharctia commented:
> Looks amazing, but the engineering lockers lacking a door will look the same like the example above it?

And what's up with the security lockers. I like the green design you've added, but unsure of their purpose.

MTandi commented:
> Looks amazing, but the engineering lockers lacking a door will look the same like the example above it?

It's just me being lazy copying the doors for the preview, please ignore.

>And what's up with the security lockers. I like the green design you've added, but unsure of their purpose.

It's the armory lockers. Right now they all have the same style, as a red security locker.

I want to change their style into a military one, like that crate that I have added. And also add icons to differentiate them in the armory by glance.

![image](https://github.com/tgstation/tgstation/assets/3625094/ae1e8976-b7de-4528-a522-2f57a41363de)

iwishforducks commented:
I think they look too shiny here. The shine on the crates was fine since the crates had so much depth, but it kinda looks a bit overbearing on the much more shallow lockers. It would be annoying to have to give new colors for these though so I understand why.

I don't like that they're taller than a tile but /shrug - that's just a personal thing

SimplySmiley123 commented:
Honestly, liking these.
A bit better looking while not overly complicated and still recognisable at a glance.

YakumoChen commented:
Really great

Only curiosity is why the tool storage lockers have locks on them now?

YakumoChen commented:
Also the HoP locker looks slightly out of place as the only non-captain head locker without the black and gold base

SomeoneYouProbablyKnow commented:
Reminds me of the cursed unity station lockers. But these are not nearly as big.

They look nice, although mixing up doors and frames makes em look kinda funky. The contrast is a bit harsh. Donno what's to be done about that.

One more thing. Basic grey locker and safe grey locker look too similar from far away now that you added top grilles both. Maybe they'll look more distinct with the light but more readability is always good.

DATA-xPUNGED commented:
i like how they're bigger actually, it was one lf the major positives of Viro's for me

MTandi committed:
2px less high, recoloring

MTandi committed:
Merge branch 'closets' of github.com:MTandi/tgstation into closets

YakumoChen commented:
Other minor nitpicks:
I know S is for science. But I think it would be better to keep the R on Research lockers for absolute clarity (colourblindless and whatnot)

Same with the cargo lockers, since captain uses C
QM should maybe have a Q instead of a C and use the Head base as well?

MTandi committed:
generic dimensions

DaydreamIQ commented:
These look amazing, though the shotgun one is a little hard to get at a glance. Is it supposed to be a shell with a pile of gunpowder next to it? Might just be a me issue there though.

EEASAS commented:
they look very nice, i just request a slight change
re-add the tiny pickaxe on the shaft miner's locker

MTandi committed:
syndicate closet dimensions

MTandi committed:
tactical door fix

MTandi commented:
Made them 2 pixels lower, changed the colors.

![dreamseeker_4S2bLx4Ytk](https://github.com/tgstation/tgstation/assets/3625094/b974bd5f-9d60-418e-a598-e2a23710d66a)

MTandi committed:
welding dimensions fix

MTandi committed:
tool closet fix

MTandi commented:
> Only curiosity is why the tool storage lockers have locks on them now?

Mistake, fixed.

>One more thing. Basic grey locker and safe grey locker look too similar from far away now that you added top grilles both. Maybe they'll look more distinct with the light but more readability is always good.

Changed.

>These look amazing, though the shotgun one is a little hard to get at a glance. Is it supposed to be a shell with a pile of gunpowder next to it? Might just be a me issue there though.

Yes, a pile of rubber balls though.

>re-add the tiny pickaxe on the shaft miner's locker

I want them to be consistent. Unfortunately, the ID icons are too big to be used on all lockers. Will consider.

MTandi committed:
icon position

SinfulBliss commented:
Why are they bigger than one tile?

excalisword commented:
Honestly, I am just glad that these don't have the weird shading that Viro's lockers had. Though I still think their shading is a little weird given how the door is darker than the top of the locker itself.

Draggeru commented:
Compared to the last locker PR these pass the Drag vibe check in terms of accessibility.

The little dots on the lockers with darker colors are a little hard to see, but thats just a me problem and I'm not necessarily the one being peddled to.

I like the lockers with the differently colored doors, it adds a layer of visual depth and helps me visually differentiate where the door starts and ends on the sprite.

jughu commented:
i wish we kept the hud icons for the lockers instead the first letter of the job

SomeoneYouProbablyKnow commented:
> i wish we kept the hud icons for the lockers instead the first letter of the job

Yeah. Some of em still have that. But the problem is we have more lockers than jobs. A lot of these use pictographs and that's very cool.

Biggest outliers are security but this change makes a good frame (🥁 ) work for anyone who'd want to change that.

OrionTheFox commented:
![image](https://github.com/tgstation/tgstation/assets/76465278/34171b57-2322-4546-9b82-bd15e212d46a)
Is the Syndie locker meant to be so tall

MTandi commented:
>Is the Syndie locker meant to be so tall Or is that image just outdated

Already fixed

MTandi committed:
closet icon and animation fix

MTandi commented:
New cabinet 

![image](https://github.com/tgstation/tgstation/assets/3625094/0c2a802e-09b5-40f6-b62a-1c0273d2d28c)

MTandi committed:
closet and guncase

MTandi committed:
wardrobes

MTandi committed:
wardrobe closets

MTandi committed:
all done

MTandi committed:
med bio

MTandi committed:
chem

MTandi committed:
chem closet

MTandi committed:
Merge branch 'master' into closets

MTandi committed:
border

MTandi committed:
Merge branch 'closets' of github.com:MTandi/tgstation into closets

MTandi committed:
R for RND

Wallemations commented:
Can we see how the wardrobes look? The lockers with all the same-colored clothes
You only posted the door, so is the door the only different part of them?

atlasle commented:
> New cabinet and guncase
> 
> 
> 
> ![image](https://github.com/tgstation/tgstation/assets/3625094/0c2a802e-09b5-40f6-b62a-1c0273d2d28c)
> 
> 
> 
> ![image](https://github.com/tgstation/tgstation/assets/3625094/6b45cf1a-94a0-4e23-b090-aecbf8362335)
> 
> 
> 
> 

that cabinet is absolutely wonderful, you need to remake the dresser in the same style

MTandi commented:
> Can we see how the wardrobes look? The lockers with all the same-colored clothes You only posted the door, so is the door the only different part of them?

Yes, they use the base gray frame, similar to the bio lockers. Only the lasertag use the rack base with bars inside, as in the armory lockers.

Wallemations commented:
> > Can we see how the wardrobes look? The lockers with all the same-colored clothes You only posted the door, so is the door the only different part of them?
> 
> Yes, they use the base gray frame, similar to the bio lockers. Only the lasertag use the rack base with bars inside, as in the armory lockers.

Can you show one/two of those?
Could just be an edit or in-game, I don't care which, just wanna see how it looks.

DAKKA-WAAAGH commented:
Damn, these sprites are top-notch! Merge ASAP

Sheits commented:
yum yum yum yum

MTandi commented:
> Can you show one/two of those? Could just be an edit or in-game, I don't care which, just wanna see how it looks.

![image](https://github.com/tgstation/tgstation/assets/3625094/b0bae07f-0c9d-4397-a826-a24eb5146423)

Wallemations commented:
Why are there scanlines on some of the icons on the doors?

MTandi commented:
> Why are there scanlines on some of the icons on the doors?

I liked how it looks on ID cards and thought that it would be nice style for white icons. Do you want me to make them flat white?

Wallemations commented:
The reason they were there on the ID cards is because those are tiny little screens right?

I don't think it makes sense on the lockers. Flat would be better imo

MTandi commented:
> The reason they were there on the ID cards is because those are tiny little screens right?
> 
> I don't think it makes sense on the lockers. Flat would be better imo

I'm not sure about the reasoning on IDs, I just thought that it would make it look like some kind of 3D objects on top of the door with a texture. Sure, I'll make them flat.

MTandi committed:
flat icons

Fikou commented:
maybe make the normal colored wardrobes still have an icon? like the W or like a t shirt or smth

MTandi commented:
> maybe make the normal colored wardrobes still have an icon? like the W or like a t shirt or smth

They are used not only for clothing.

Wallemations commented:
> > maybe make the normal colored wardrobes still have an icon? like the W or like a t shirt or smth
> 
> 
> 
> They are used not only for clothing.

Either way they are still rather bare, and my own main concern with that is the grey one specifically, as it looks pretty much 1:1 with the normal locker.

MTandi commented:
Well they are bare by design. I can remove the gray locker and replace it with the generic one, but that will require path updates. Please suggest the solution.

MTandi committed:
WWWWWWWWW

MTandi committed:
grayer mixed

MTandi committed:
cardboard fix
